:farmer: ros2cli tests failing in rolling zenoh because zennoh router isn't working

### Generated by Generative AI

_No response_

### Operating System:

Ubuntu noble

### ROS version or commit hash:

Rolling

### RMW implementation (if applicable):

rmw_zenoh

### RMW Configuration (if applicable):

_No response_

### Client library (if applicable):

_No response_

### 'ros2 doctor --report' output

_No response_

### Steps to reproduce issue

1. Go to http://build.ros2.org/job/Rci__nightly-zenoh_ubuntu_noble_amd64
2. Build a job
3. See the tests failing

### Expected behavior

To build without test regressions

### Actual behavior

Building with regressions

### Additional information

### Reference builds
- [Rci__nightly-zenoh_ubuntu_noble_amd64#372](http://build.ros2.org/job/Rci__nightly-zenoh_ubuntu_noble_amd64/372)

### Failing tests
<details><summary>Show all tests failing</summary>

- projectroot.test_cli_remapping
- ros2cli.ros2cli.test.test_ros2cli_daemon.test_get_name
- ros2cli.ros2cli.test.test_ros2cli_daemon.test_get_namespace
- ros2cli.ros2cli.test.test_ros2cli_daemon.test_get_node_names_and_namespaces
- ros2cli.ros2cli.test.test_ros2cli_daemon.test_get_node_names_and_namespaces_with_enclaves
- ros2cli.ros2cli.test.test_ros2cli_daemon.test_get_topic_names_and_types
- ros2cli.ros2cli.test.test_ros2cli_daemon.test_get_service_names_and_types
- ros2cli.ros2cli.test.test_ros2cli_daemon.test_get_action_names_and_types
- ros2cli.ros2cli.test.test_ros2cli_daemon.test_get_publisher_names_and_types_by_node
- ros2cli.ros2cli.test.test_ros2cli_daemon.test_get_subscriber_names_and_types_by_node
- ros2cli.ros2cli.test.test_ros2cli_daemon.test_get_service_names_and_types_by_node
- ros2cli.ros2cli.test.test_ros2cli_daemon.test_get_client_names_and_types_by_node
- ros2cli.ros2cli.test.test_ros2cli_daemon.test_get_action_server_names_and_types_by_node
- ros2cli.ros2cli.test.test_ros2cli_daemon.test_get_action_client_names_and_types_by_node
- ros2cli.ros2cli.test.test_ros2cli_daemon.test_get_publishers_info_by_topic
- ros2cli.ros2cli.test.test_ros2cli_daemon.test_get_subscriptions_info_by_topic
- ros2cli.ros2cli.test.test_ros2cli_daemon.test_get_clients_info_by_service
- ros2cli.ros2cli.test.test_ros2cli_daemon.test_get_servers_info_by_service
- ros2cli.ros2cli.test.test_ros2cli_daemon.test_count_publishers
- ros2cli.ros2cli.test.test_ros2cli_daemon.test_count_subscribers
- ros2cli.ros2cli.test.test_ros2cli_daemon.test_count_clients
- ros2cli.ros2cli.test.test_ros2cli_daemon.test_count_services
- test_cli_remapping.TestCLIRemapping.test_namespace_replacement[/tmp/ws/build_isolated/test_cli_remapping/name_maker_rclcpp]
- test_cli_remapping.TestCLIRemapping.test_node_name_replacement[/tmp/ws/build_isolated/test_cli_remapping/name_maker_rclcpp]
- test_cli_remapping.TestCLIRemapping.test_service_replacement[/tmp/ws/build_isolated/test_cli_remapping/name_maker_rclcpp]
- test_cli_remapping.TestCLIRemapping.test_topic_and_service_replacement[/tmp/ws/build_isolated/test_cli_remapping/name_maker_rclcpp]
- test_cli_remapping.TestCLIRemapping.test_topic_replacement[/tmp/ws/build_isolated/test_cli_remapping/name_maker_rclcpp]
- test_cli_remapping.TestCLIRemapping.test_namespace_replacement[/tmp/ws/src/ros2/system_tests/test_cli_remapping/test/name_maker.py]
- test_cli_remapping.TestCLIRemapping.test_node_name_replacement[/tmp/ws/src/ros2/system_tests/test_cli_remapping/test/name_maker.py]
- test_cli_remapping.TestCLIRemapping.test_service_replacement[/tmp/ws/src/ros2/system_tests/test_cli_remapping/test/name_maker.py]
- test_cli_remapping.TestCLIRemapping.test_topic_and_service_replacement[/tmp/ws/src/ros2/system_tests/test_cli_remapping/test/name_maker.py]
- test_cli_remapping.TestCLIRemapping.test_topic_replacement[/tmp/ws/src/ros2/system_tests/test_cli_remapping/test/name_maker.py]
</details>

### Log Output 1
```
test/test_ros2cli_daemon.py:131: in daemon_node
    pytest.fail(
E   Failed: daemon failed to discover /test/test_node
---------------------------- Captured stdout setup -----------------------------
[2m2026-03-17T15:26:40.734330Z[0m [33m WARN[0m ThreadId(06) [2mzenoh::net::runtime::orchestrator[0m[2m:[0m Scouting delay elapsed before start conditions are met.
---------------------------- Captured stderr setup -----------------------------
[WARN] [1773761200.734402301] [rmw_zenoh_cpp]: Unable to connect to a Zenoh router. Have you started a router with `ros2 run rmw_zenoh_cpp rmw_zenohd`?
[WARN] [1773761200.734427360] [rmw_zenoh_cpp]: Unable to connect to a Zenoh router after 1 attempt(s). Please ensure that a Zenoh router is running and can be reached. You may increase the number of attempts to check for a router by setting the ZENOH_ROUTER_CHECK_ATTEMPTS environment variable. Proceeding with initialization but other peers will not discover or receive data from peers in this session until a router is started.
_____________________ ERROR at setup of test_get_namespace _____________________
test/test_ros2cli_daemon.py:131: in daemon_node
    pytest.fail(
E   Failed: daemon failed to discover /test/test_node
_____________ ERROR at setup of test_get_node_names_and_namespaces _____________
test/test_ros2cli_daemon.py:131: in daemon_node
    pytest.fail(
E   Failed: daemon failed to discover /test/test_node
______ ERROR at setup of test_get_node_names_and_namespaces_with_enclaves ______
test/test_ros2cli_daemon.py:131: in daemon_node
    pytest.fail(
E   Failed: daemon failed to discover /test/test_node
_______________ ERROR at setup of test_get_topic_names_and_types _______________

```

### log Ouput 2
```
======================================================================
1: FAIL: test_namespace_replacement[/tmp/ws/src/ros2/system_tests/test_cli_remapping/test/name_maker.py] (test_cli_remapping.TestCLIRemapping.test_namespace_replacement[/tmp/ws/src/ros2/system_tests/test_cli_remapping/test/name_maker.py])
1: ----------------------------------------------------------------------
1: Traceback (most recent call last):
1:   File "/tmp/ws/src/ros2/system_tests/test_cli_remapping/test/test_cli_remapping.py", line 129, in test_namespace_replacement
1:     self.assertIn(namespace_replacement, self.get_topics())
1: AssertionError: '/ns/s5051_15_36_28/relative/name' not found in ['/parameter_events', '/rosout']
1: 
1: ======================================================================
1: FAIL: test_node_name_replacement[/tmp/ws/src/ros2/system_tests/test_cli_remapping/test/name_maker.py] (test_cli_remapping.TestCLIRemapping.test_node_name_replacement[/tmp/ws/src/ros2/system_tests/test_cli_remapping/test/name_maker.py])
1: ----------------------------------------------------------------------
1: Traceback (most recent call last):
1:   File "/tmp/ws/src/ros2/system_tests/test_cli_remapping/test/test_cli_remapping.py", line 138, in test_node_name_replacement
1:     self.assertIn(node_name_replacement, self.node.get_node_names())
1: AssertionError: 'node_2213_15_36_28' not found in ['test_cli_remapping']
1: 
1: ======================================================================
1: FAIL: test_service_replacement[/tmp/ws/src/ros2/system_tests/test_cli_remapping/test/name_maker.py] (test_cli_remapping.TestCLIRemapping.test_service_replacement[/tmp/ws/src/ros2/system_tests/test_cli_remapping/test/name_maker.py])
1: ----------------------------------------------------------------------
1: Traceback (most recent call last):
1:   File "/tmp/ws/src/ros2/system_tests/test_cli_remapping/test/test_cli_remapping.py", line 168, in test_service_replacement
1:     self.assertIn(service_replacement, self.get_services())
1: AssertionError: '/remapped/s2481_15_36_28' not found in ['/test_cli_remapping/describe_parameters', '/test_cli_remapping/get_parameter_types', '/test_cli_remapping/get_parameters', '/test_cli_remapping/get_type_description', '/test_cli_remapping/list_parameters', '/test_cli_remapping/set_parameters', '/test_cli_remapping/set_parameters_atomically']
1
```

### Flakyness 15 days
job_name|last_fail|first_fail|build_count|failure_count|failure_percentage
|--|--|--|--|--|--|
Rci__nightly-zenoh_ubuntu_noble_amd64|2026-03-17|2026-03-14|4|4|100.0

These are two different types of tests failing but both came from the same error, zenoh daemon isn't starting so the second type of tests aren't finding the respective name topic.
This have been happening for a lot of time, so in the log I can't check from when it began to happen.

CC: @Crola1702

--- a/ros2cli/package.xml
+++ b/ros2cli/package.xml
@@ -23,6 +23,7 @@
   <exec_depend>rclpy</exec_depend>
   <exec_depend>fzf</exec_depend>
 
+  <test_depend>rmw_test_fixture_implementation</test_depend>
   <test_depend>ament_copyright</test_depend>
   <test_depend>ament_flake8</test_depend>
   <test_depend>ament_pep257</test_depend>

--- a/ros2cli/test/test_ros2cli_daemon.py
+++ b/ros2cli/test/test_ros2cli_daemon.py
@@ -27,6 +27,13 @@ from ros2cli.node.daemon import is_daemon_running
 from ros2cli.node.daemon import shutdown_daemon
 from ros2cli.node.daemon import spawn_daemon
 
+try:
+    from rmw_test_fixture_implementation import rmw_test_isolation_start
+    from rmw_test_fixture_implementation import rmw_test_isolation_stop
+    _HAS_RMW_ISOLATION = True
+except ImportError:
+    _HAS_RMW_ISOLATION = False
+
 import test_msgs.action
 import test_msgs.msg
 import test_msgs.srv
@@ -54,6 +61,16 @@ TEST_SERVICE_NAME = '/test/service'
 TEST_SERVICE_TYPE = 'test_msgs/srv/Empty'
 TEST_ACTION_NAME = '/test/action'
 TEST_ACTION_TYPE = 'test_msgs/action/Fibonacci'
+
+
+@pytest.fixture(autouse=True, scope='session')
+def rmw_isolation():
+    """Start RMW isolation before spawning the daemon."""
+    if _HAS_RMW_ISOLATION:
+        rmw_test_isolation_start()
+    yield
+    if _HAS_RMW_ISOLATION:
+        rmw_test_isolation_stop()
 
 
 @pytest.fixture(autouse=True, scope='module')

